DESCRIPTION

Git API does not declare GitAPIException call() and related cleanups

All commands should throw a GitAPIException so new exceptions can be
added without breaking the builds of old code, i.e. anyone that calls
a Git API should catch GitAPIException and not just the currently known
exceptions.

Now the only checked exceptions on Git API calls are GitException and
subclasses of it. New checked exceptions that are subclasses of
GitException may be added without breaking the API.

Javadoc for GitAPIException is declared on GitCommand and
inherited to subclasses. JGitInternalException is not explicitly
documented anymore.

Unfortunately this change itself breaks the API. The intention is
that it shall be possible to add new checked subclasses of
GitAPIException without breaking the API.

Bug: 366914
EGit-Change-Id: I50380f13fc82c22d0036f47c7859cc3a77e767c5
Change-Id: I50380f13fc82c22d0036f47c7859cc3a77e767c5
Signed-off-by: Douglas Trujillo xxx@xxx.xxx


COMMENTS

author: Gerrit Code Review
date: 2012-05-30 20:00:33.000000000

Your change could not be merged due to a path conflict.

Please merge (or rebase) the change locally and upload the resolution for review.

-------------------------------------
author: Juliet Cantu
date: 2012-05-30 20:10:21.000000000

Uploaded patch set 11.

-------------------------------------
author: Juliet Cantu
date: 2012-05-30 20:21:01.000000000

Patch Set 11:

rebased

-------------------------------------
author: Douglas Trujillo
date: 2012-05-30 20:21:17.000000000

Patch Set 11: Verified; Looks good to me, approved; IP review completed

got rebased twice, tested locally

-------------------------------------
author: Douglas Trujillo
date: 2012-05-30 20:21:20.000000000

Change has been successfully merged into the git repository.

-------------------------------------
